thunder/layout/fixed.py

```diff
@@ -0,0 +1,49 @@
+from typing import Dict, Optional, Sequence, Union
+
+from more_itertools import collapse, zip_equal
+
+from .split import SingleSplit, Split
+
+
+class FixedSplit(Split):
+    def __init__(self, splits: Sequence,
+                 names: Optional[Sequence[str]] = None):
+        """
+        Creates experiment layout from given split.
+        Parameters
+        ----------
+        splits: Sequence
+            Split of data.
+        names: Optional[Sequence[str]]
+            Names of folds, e.g. 'train', 'val', test'.
+        """
+        if names is not None:
+            # TODO
+            assert len(set(names)) == len(names)
+            assert len(splits[0]) == len(names)
+
+        self.entries = sorted(set(collapse(splits)))
+        self.splits = [tuple(fold) for fold in splits]
+        self.names = names
+        self.fold: Optional[int] = None
+
+
+class FixedSingleSplit(SingleSplit):
```

Comment:
why do you need this class? isn't this the same as FixedSplit with a single split?